script.js

```diff
@@ -1,406 +1,399 @@
+        <p><b>Cuisine:</b> ${capitalizedCuisineType}</p>
+        <p><b>Cooking time:</b> ${recipe.totalTime} minutes</p>
+        <p><b>Ingredients:</b></p>
+        <ul>
+          ${recipe.ingredients
+            .map((ingredient) => `<li>${ingredient}</li>`)
+            .join("")}
+        </ul>
+        <p class="recipe-link"><a href="${
+          recipe.url
+        }" target="_blank">Link to cooking method</a>
+        </p>
+      </div>
+    </div>
+    `
+  })
+}
+
+//Initial rendering of all recipes
+showRecipes(recipes)
+
+//Function random
+const getRandomValue = () => {
+  container.innerHTML = ""
+
+  recipes[Math.floor(Math.random() * recipes.length)]
+  const recipe = recipes[Math.floor(Math.random() * recipes.length)]
+  container.innerHTML += `
+  <div class="recipe-card">
+    <img src="${recipe.image}"/>
+    <div class="recipe-text">
+      <h3>${recipe.name}</h3>
+      <p><b>Cuisine:</b> ${recipe.cuisineType}</p>
+      <p><b>Cooking time</b>: ${recipe.totalTime} minutes</p>
+      <p><b>Ingredients:</b></p>
+      <ul>
+        ${recipe.ingredients
+          .map((ingredient) => `<li>${ingredient}</li>`)
+          .join("")}
+      </ul>
+      <p class="recipe-link"><a href="${
+        recipe.url
+      }" target="_blank">Link to cooking method</a>
+      </div>
+  </div>
+  `
+}
+
+//Function to show favorites
+const showFavorites = () => {
+  const favRecipes = recipes.filter((recipe) => recipe.favorite === true)
+  showRecipes(favRecipes)
+}
+
+//Function to sorting descending
+const filteredRecipesByDescending = () => {
+  recipes.sort((a, b) => a.totalTime - b.totalTime)
+  showRecipes(recipes)
+}
+
+//Function to sorting ascending
+const filteredRecipesByAscending = () => {
+  recipes.sort((a, b) => b.totalTime - a.totalTime)
+  showRecipes(recipes)
+}
+
+//Function to filter on search input based on name or ingredient
+const filterRecipesBySearchInput = (searchInputValue) => {
+  const filteredRecipes = recipes.filter(
+    (recipe) =>
+      recipe.name.toLowerCase().includes(searchInputValue.toLowerCase()) ||
+      recipe.ingredients.some((ingredient) =>
+        ingredient.toLowerCase().includes(searchInputValue.toLowerCase())
+      )
+  )
+  showRecipes(filteredRecipes)
+}
+
+//Function to filter on cuisineType
+const filterRecipesByCuisine = (cuisineType) => {
+  const filteredRecipes = recipes.filter((recipe) =>
+    recipe.cuisineType.includes(cuisineType)
+  )
+  //Call the showRecipes funciton with the filtered recipes
+  showRecipes(filteredRecipes)
+}
+
+//Event listeners for the filter dropdown
+filterRecipes.addEventListener("change", (event) => {
+  const selectedCuisine = event.target.value
+
+  // if selected value = all -> then show all recipes otherwise filter by cuisine type
+  if (selectedCuisine === "all") {
+    showRecipes(recipes)
+  } else {
+    filterRecipesByCuisine(selectedCuisine)
+  }
+})
+
+//Event listener for favorites
+favoriteButton.addEventListener("click", (event) => {
```

Comment:
Normal buttons have no default behavior, therefore the preventdefault is not necessery for normal buttons, they do not reload the page on default like the submit button would do. Your eventlisteners could be more compact and sleaker by removing this :D